stubs/emoji/emoji/core.pyi

```diff
@@ -1,17 +1,38 @@
+from collections.abc import Callable
 from typing import Pattern
 from typing_extensions import Literal
 
 _DEFAULT_DELIMITER: str
 
+class _DeprecatedParameter: ...
+
 def emojize(
     string: str,
-    use_aliases: bool = ...,
+    use_aliases: bool | type[_DeprecatedParameter] = ...,
     delimiters: tuple[str, str] = ...,
     variant: Literal["text_type", "emoji_type", None] = ...,
     language: str = ...,
+    version: float | int | None = ...,
+    handle_version: str | Callable[[str, dict[str, str]], str] | None = ...,
+) -> str: ...
+def demojize(
+    string: str,
+    use_aliases: bool | type[_DeprecatedParameter] = ...,
+    delimiters: tuple[str, str] = ...,
+    language: str = ...,
+    version: float | int | None = ...,
+    handle_version: str | Callable[[str, dict[str, str]], str] | None = ...,
+) -> str: ...
+def replace_emoji(
+    string: str,
+    replace: str | Callable[[str, dict[str, str]], str] = ...,
+    language: str | type[_DeprecatedParameter] = ...,
+    version: float | int | None = ...,
 ) -> str: ...
-def demojize(string: str, use_aliases: bool = ..., delimiters: tuple[str, str] = ..., language: str = ...) -> str: ...
-def get_emoji_regexp(language: str = ...) -> Pattern[str]: ...
-def emoji_lis(string: str, language: str = ...) -> list[dict[str, int | str]]: ...
-def distinct_emoji_lis(string: str) -> list[str]: ...
-def emoji_count(string: str) -> int: ...
+def get_emoji_regexp(language: str | None = ...) -> Pattern[str]: ...
+def emoji_lis(string: str, language: str | type[_DeprecatedParameter] = ...) -> list[dict[str, int | str]]: ...
```

Comment:
You could consider returning a list of `TypedDict`s here: the type associated with the `location` key will always be `int`; the type associated with the `emoji` key will always be `str`. https://docs.python.org/3/library/typing.html#typing.TypedDict.

Same point applies for the `emoji_list` function below.